service/matching/matching_engine_test.go

```diff
@@ -369,28 +369,102 @@ func (s *matchingEngineSuite) TestPollWorkflowTaskQueuesEmptyResultWithShortCont
+	const pollCount = 10
+	for i := 0; i < pollCount; i++ {
+		if taskType == enumspb.TASK_QUEUE_TYPE_ACTIVITY {
+			pollResp, err := s.matchingEngine.PollActivityTaskQueue(callContext, &matchingservice.PollActivityTaskQueueRequest{
+				NamespaceId: namespaceId,
+				PollRequest: &workflowservice.PollActivityTaskQueueRequest{
+					TaskQueue: taskQueue,
+					Identity:  identity,
+				},
+			}, metrics.NoopMetricsHandler)
+			s.NoError(err)
+			s.Equal(emptyPollActivityTaskQueueResponse, pollResp)
+
+			taskQueueType = enumspb.TASK_QUEUE_TYPE_ACTIVITY
+		} else {
+			resp, err := s.matchingEngine.PollWorkflowTaskQueue(callContext, &matchingservice.PollWorkflowTaskQueueRequest{
+				NamespaceId: namespaceId,
+				PollRequest: &workflowservice.PollWorkflowTaskQueueRequest{
+					TaskQueue: taskQueue,
+					Identity:  identity,
+				},
+			}, metrics.NoopMetricsHandler)
+			s.NoError(err)
+			s.Equal(emptyPollWorkflowTaskQueueResponse, resp)
+
+			taskQueueType = enumspb.TASK_QUEUE_TYPE_WORKFLOW
+		}
+		select {
+		case <-callContext.Done():
+			s.FailNow("Call context has expired.")
+		default:
+		}
+		// check the poller information
+		descResp, err := s.matchingEngine.DescribeTaskQueue(context.Background(), &matchingservice.DescribeTaskQueueRequest{
+			NamespaceId: namespaceId,
+			DescRequest: &workflowservice.DescribeTaskQueueRequest{
+				TaskQueue:              taskQueue,
+				TaskQueueType:          taskQueueType,
+				IncludeTaskQueueStatus: false,
+			},
+		})
+		s.NoError(err)
+		s.Equal(1, len(descResp.DescResponse.Pollers))
+		s.Equal(identity, descResp.DescResponse.Pollers[0].GetIdentity())
+		s.NotEmpty(descResp.DescResponse.Pollers[0].GetLastAccessTime())
+		s.Nil(descResp.DescResponse.GetTaskQueueStatus())
+	}
+	s.EqualValues(1, s.taskManager.getQueueManager(tlID).RangeID())
+}
+
 func (s *matchingEngineSuite) TestOnlyUnloadMatchingInstance() {
+	s.mockMatchingClient.EXPECT().ForceLoadTaskQueuePartition(gomock.Any(), gomock.Any()).
+		Return(&matchingservice.ForceLoadTaskQueuePartitionResponse{WasUnloaded: true}, nil).MaxTimes(3)
+
 	prtn := newRootPartition(
 		uuid.New(),
 		"makeToast",
 		enumspb.TASK_QUEUE_TYPE_ACTIVITY)
-	tqm, err := s.matchingEngine.getTaskQueuePartitionManager(context.Background(), prtn, true, loadCauseUnspecified)
+	tqm, _, err := s.matchingEngine.getTaskQueuePartitionManager(context.Background(), prtn, true, loadCauseUnspecified)
 	s.Require().NoError(err)
 
 	tqm2 := s.newPartitionManager(prtn, s.matchingEngine.config)
 
 	// try to unload a different tqm instance with the same taskqueue ID
 	s.matchingEngine.unloadTaskQueuePartition(tqm2, unloadCauseUnspecified)
 
-	got, err := s.matchingEngine.getTaskQueuePartitionManager(context.Background(), prtn, true, loadCauseUnspecified)
+	s.mockMatchingClient.EXPECT().ForceLoadTaskQueuePartition(gomock.Any(), gomock.Any()).
+		Return(&matchingservice.ForceLoadTaskQueuePartitionResponse{WasUnloaded: true}, nil).MaxTimes(3)
```

Comment:
can you just do this in SetupTest (e.g. line 173 below the other "expected" matching client calls), and then you don't need to do it on every test?

Reply:
Sure. I may leave a few tests where the expectation is more valuable.

Reply:
This wasn't done? I don't see any change in SetupTest and I still see a lot of EXPECT calls?